### ISSUE TITLE
Broken if-statement

### ISSUE THREAD

**Issue by mohamed-barakat** [homalg-project/CAP_project#1436]
The condition in this if-statement

https://github.com/homalg-project/CAP_project/blob/039bfd462710bd481024c1b501ad9681e6160c09/CompilerForCAP/gap/InferDataTypes.gi#L645

should include `and not data_type = "list_with_unknown_element_type"`.

**Comment by zickgraf** [comment]
Did you give this a try? I don't expect this to work, i.e. I expect that things will simply break later on.

**Comment by mohamed-barakat** [comment]
No, I did not. This record access was caused by another error I had, so I stopped following it.
